zebra-rpc/src/methods.rs

```diff
@@ -2631,31 +2633,57 @@ impl OrchardTrees {
     }
 }
 
-/// Check if provided height range is valid for address indexes.
-fn check_height_range(start: Height, end: Height, chain_height: Height) -> Result<()> {
-    if start == Height(0) || end == Height(0) {
-        return Err(ErrorObject::owned(
-            ErrorCode::InvalidParams.code(),
-            format!("start {start:?} and end {end:?} must both be greater than zero"),
-            None::<()>,
-        ));
-    }
+/// Build a valid height range from the given optional start and end numbers.
+///
+/// # Parameters
+///
+/// - `start`: Optional starting height. If not provided, defaults to 0.
+/// - `end`: Optional ending height. A value of 0 or absence of a value indicates to use `chain_height`.
+/// - `chain_height`: The maximum permissible height.
+///
+/// # Returns
+///
+/// A `RangeInclusive<Height>` from the clamped start to the clamped end.
+///
+/// # Errors
+///
+/// Returns an error if the computed start is greater than the computed end.
+fn build_height_range(
+    start: Option<u32>,
+    end: Option<u32>,
+    chain_height: Height,
+) -> Result<RangeInclusive<Height>> {
+    // Convert optional values to Height, using 0 (as Height(0)) when missing.
+    let start = start.map(Height).unwrap_or(Height(0));
+    // If start is above chain_height, clamp it to chain_height.
+    let start = if start > chain_height {
+        chain_height
+    } else {
+        start
+    };
```

Comment:
Nitpick/Optional/Subjective:

```suggestion
    // Convert optional values to Height, using 0 (as Height(0)) when missing.
    // If start is above chain_height, clamp it to chain_height.
    let start = Height(start.unwrap_or(0)).min(chain_height);
```

```